admin/controllers/pupil.js

```diff
@@ -59,33 +65,14 @@ const postAddPupil = async (req, res, next) => {
     pinExpired: false
   })
   try {
-    await pupil.validate()
-    if (validationError.hasError()) {
-      throw new Error('custom validation error')
-    }
+    const pupilData = req.body
+    await validatePupil(pupil, pupilData)
   } catch (error) {
-    req.breadcrumbs(res.locals.pageTitle)
-    if (error.message !== 'custom validation error') {
-      // Mongoose error
-      // At this point we have validated the schema and may or may not have anything in validationError
-      // So = combine all validation errors into one
-      const combinedValidationError = errorConverter.fromMongoose(error, addPupilErrorMessages, validationError)
-      // error fixup: if the mongoose schema bails out on the dob field - we should make sure we have some
-      // actual html fields that have an error.  If we do, we can ditch the mongoose error as being superfluous.
-      if (combinedValidationError.isError('dob') && (combinedValidationError.isError('dob-day') || combinedValidationError.isError('dob-month') || combinedValidationError.isError('dob-year'))) {
-        combinedValidationError.removeError('dob')
-      }
-      return res.render('school/add-pupil', {
-        school: school.toJSON(),
-        formData: req.body,
-        error: combinedValidationError,
-        breadcrumbs: req.breadcrumbs()
-      })
-    }
+    Object.keys(error.errors).forEach((e) => { error.errors[e] = error.errors[e].replace(/\\/g, '') })
```

Comment:
This code seems far too low-level to be in a controller.  Why?

Reply:
@jon-shipley It won't live in the controller as soon as I refactor it as part of the next PR